fenix/app/src/main/java/org/mozilla/fenix/components/appstate/AppAction.kt

```diff
@@ -238,4 +239,27 @@ sealed class AppAction : Action {
          */
         data class RemoveFromProductAnalysed(val productPageUrl: String) : ShoppingAction()
     }
+
+    /**
+     * [Actions]s related to interactions with the [AwesomeBar].
+     */
+    sealed class AwesomeBarAction : AppAction() {
```

Comment:
I could also see these actions moving to `feature-fxsuggest`, and exposing an `FxSuggestAwesomeBarStateReducer` type from that component that Fenix's `AppStoreReducer` would call—basically, copying the `ShoppingStateReducer` pattern.

If we did that, I think we could have `feature-fxsuggest` emit facts for clicks and impressions again!

I kinda like that, but also want to make sure I'm not overthinking it, or copying a pattern that doesn't make sense. WDYT @jonalmeida?

Reply:
Oh, that's a cool idea! It's up to you if you want to do that in this patch or a future iteration (thinking about time and release commitments).

It's also follows the pattern that `SyncedTabsStorageSuggestionProvider` has as well for emitting facts from the provider (minus the reducer part).